### PR TITLE
Added model for telemetry service resources

### PR DESCRIPTION
**Pull request by shivacharanms**

### CODE REVIEW

File: lib-dmtf/model/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	MetricProperties    []string `json:"MetricProperties,omitempty"`
+	Oem                 *Oem     `json:"Oem,omitempty"`
+}
+
+type MetricLink struct {
+	Oem      *Oem  `json:"Oem,omitempty"`
+	Triggers []Oid `json:"Triggers,omitempty"`
+}
+
+type Oid struct {
```
Comment (Bharath-KKB) [inline lib-dmtf/model/telemetry.go:98]: we should not redfine this. Confirm this is defined elsewhere in the same package and if so remove this.

File: lib-dmtf/model/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	ODataID string `json:"@odata.id"`
+}
+
+type MetricReports struct {
```
Comment (Bharath-KKB) [inline lib-dmtf/model/telemetry.go:102]: OData.Id, OData.Type Id and Name are mandatory attributes. CHeck this for all other schemas as well

Comment (Bharath-KKB) [reply lib-dmtf/model/telemetry.go:102]: Looks like we cannot just rely on DSP2046 alone. Have to consider the schema files as well.(for types/schemas

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
cross check with schemas as well. DSP2046 doesnt seem to cover all the params.

File: lib-dmtf/model/telemetry.go (hunk 1 above)
Comment (shivacharanms) [reply lib-dmtf/model/telemetry.go:98]: As a separate struct, this is not present.

File: lib-dmtf/model/telemetry.go (hunk 2 above)
Comment (shivacharanms) [reply lib-dmtf/model/telemetry.go:102]: addressed

**Review by jeevan-kamkar — APPROVED** [review]

**Review by Bharath-KKB — APPROVED** [review]

**Pull request merged**